locallib.php

```diff
@@ -60,6 +60,29 @@ function tool_crawler_link($url, $label, $redirect = '', $labelishtml = false) {
     return $html;
 }
 
+/**
+ * Renders a priority level integer as readable HTML
+ *
+ * @param int $priority the priority level
+ * @return string HTML snippet which can be used in output.
+ */
+function tool_crawler_priority_level($priority) {
+
+    $html = '';
+    if ($priority == TOOL_CRAWLER_PRIORITY_DEFAULT) {
+        $html = html_writer::div(get_string('defaultpriority', 'tool_crawler'), 'priority');
+    }
+    if ($priority == TOOL_CRAWLER_PRIORITY_NORMAL) {
+        $html = html_writer::div(get_string('normalpriority', 'tool_crawler'), 'priority');
+    }
+    if ($priority == TOOL_CRAWLER_PRIORITY_UPDATE_EVENT) {
+        $html = html_writer::div(get_string('eventupdatedpriority', 'tool_crawler'), 'priority');
+    }
+    if (!$html) {
+        $html = html_writer::div(get_string('custompriority', 'tool_crawler') . $priority, 'priority');
```

Comment:
This is generally better as:

$html = html_writer::div(get_string('custompriority', 'tool_crawler', $priority), 'priority');

$string['custompriority'] = 'Custom priority: {$a}';

Reply:
Thank you, added that in now